src/vs/base/common/errors.ts

```diff
@@ -258,7 +265,7 @@ export class ErrorNoTelemetry extends Error {
 	}
 
 	public static isErrorNoTelemetry(err: Error): err is ErrorNoTelemetry {
```

Comment:
Can we implement this method differently without having to change the name of the error, e.g. by adding another property that is private to the error?

Reply:
We tried that but found when errors got serialized across processes that it would be only the base error properties that were left and therefore we had to use one of the three error properties and figured name was the least important one. @sbatten and I have tried many solutions but our architecture is quite complex so ends up breaking the ErrorNoTelemetry one way or another. It seems the name probing is unique to FileSystemErrors